api/src/main/java/jakarta/config/Config.java

```diff
@@ -118,6 +119,21 @@
      */
     <T> Optional<List<T>> asList(Class<T> type);
 
+    /**
+     * Register a change consumer for this config node. The consumer would only be called if a change in configuration impacts
+     * the current node or its sub-nodes.
+     * <p>
+     * Example 1: If this node is {@code server.host} and {@code server.host} is modified, the consumer would be triggered,
+     *  in case {@code server.port} is modified, the consumer would not be triggered.
+     * <p>
+     * Example 2: If this node is {@code server} and either {@code server.host} or {@code server.port} is modified, the
+     * consumer would be triggered.
+     *
+     * @param changeConsumer consumer of changes, the config provided to the consumer is the same node as the current node,
+     *                          but with values from the changed config source
+     */
+    void onChange(Consumer<Config> changeConsumer);
```

Comment:
Should we allow the subscriber to reject a change as invalid, so that _n_ subscribers do not end up with inconsistent views of configuration? Or, for another case, if a subscriber to an outer node determines that a new configuration is invalid, but an inner node subscriber thought it was valid, how do we handle that?

Reply:
I think there are too many edge cases to do this. I would prefer that if the config changes and you get wrong values, it is your responsibility to handle it (log it, just ignore the new config etc.).
What I see as a main issue here is that if I get a new config and there are 10 listeners, 8 may process it and 2 mark it as invalid. There is no way to rollback the 8 listeners that already processed the new values